File: problem2/stocktick.py
class StockTick:
   def __init__(self, text_line=None, date="", time="", price=0.0, bid=0.0, ask=0.0, units=0):
      if text_line != None:
         tokens = text_line.split(",")
         self.date = tokens[0]
         self.time = tokens[1]
         try:
            self.price = float(tokens[2])
            self.bid = float(tokens[3])
            self.ask = float(tokens[4])
            self.units = int(tokens[5])
         except:
            self.price = 0.0
            self.bid = 0.0
            self.ask = 0.0
            self.units = 0
      else:
         self.date = date
         self.time = time
         self.price = price
         self.bid = bid
         self.ask = ask
         self.units = units

File: problem2/Stock/src/main/scala/com/stock/stocktick.scala
package com.stock

class StockTick (val text_line:String="", var date:String="", var time:String="", 
    var price:Double=0.0, var bid:Double=0.0, var ask:Double=0.0, var units:Int=0) extends java.io.Serializable
{
  var tokens : Array[String] = new Array[String](4)
  if (text_line.length() !=0) {
         tokens = text_line.split(",")
         date = tokens(0)
         time = tokens(1)

         try {
            price = tokens(2).toDouble
            bid = tokens(3).toDouble
            ask = tokens(4).toDouble
            units = tokens(5).toInt
         } catch {
           case ex : Exception => {
                 price=0.0
                 bid = 0.0
                 ask = 0.0
                 units = 0
           }
          }
  }
  else{
         date = date
         time = time
         price = price
         bid = bid
         ask = ask
         units = units
  }
}
